lib/util.cpp

```diff
@@ -542,3 +542,38 @@ bool process_exists(int pid) {
 }
 
 #endif
+
+unsigned int random_int() {
```

Comment:
I suggest to change signature function to something like this:
```bool random_int(unsigned int& value)```
And return the random value using the output parameter, and return `true` on success and `false` in case of an error

Reply:
Also, then you need to adjust `crypt_prog.cpp` with the new function signature